wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -388,6 +396,31 @@ private void cleanupFiles() {
         }
     }
 
+    private void initNotificationManager() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            final Stopwatch watch = Stopwatch.createStarted();
+            final NotificationManager nm = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+
+            final NotificationChannel received = new NotificationChannel(Constants.NOTIFICATION_CHANNEL_ID_RECEIVED,
+                    getString(R.string.notification_channel_received_name), NotificationManager.IMPORTANCE_DEFAULT);
+            received.setSound(Uri.parse("android.resource://" + getPackageName() + "/" + R.raw.coins_received),
+                    new AudioAttributes.Builder().setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+                            .setLegacyStreamType(AudioManager.STREAM_NOTIFICATION)
+                            .setUsage(AudioAttributes.USAGE_NOTIFICATION_EVENT).build());
+            nm.createNotificationChannel(received);
+
+            final NotificationChannel ongoing = new NotificationChannel(Constants.NOTIFICATION_CHANNEL_ID_ONGOING,
+                    getString(R.string.notification_channel_ongoing_name), NotificationManager.IMPORTANCE_LOW);
+            nm.createNotificationChannel(ongoing);
+
+            final NotificationChannel important = new NotificationChannel(Constants.NOTIFICATION_CHANNEL_ID_IMPORTANT,
+                    getString(R.string.notification_channel_important_name), NotificationManager.IMPORTANCE_HIGH);
+            nm.createNotificationChannel(important);
```

Comment:
Yes, but does it have an advantage? I think as long as the collection of channels isn't dynamic the single form is more appropriate.